conanfile.py

```diff
@@ -135,21 +136,33 @@ def _gather_licenses(self, bootstrap):
             f" --with-license-file --no-license-path --output-file=package_licenses.txt"
         )
 
-    def build(self):
+    def source(self):
         replace_in_file(self, "embedded_python.cmake", "${self.pyversion}", str(self.pyversion))
 
+        if self.settings.os != "Windows":
+            build_helper = UnixLikeBuildHelper(self)
+            build_helper.get_source()
+
+    def generate(self):
+        prefix = pathlib.Path(self.build_folder) / "embedded_python"
+        if self.settings.os != "Windows":
+            build_helper = UnixLikeBuildHelper(self)
+            build_helper.generate(prefix)
+
+    def build(self):
         prefix = pathlib.Path(self.build_folder) / "embedded_python"
         if self.settings.os == "Windows":
-            build_helper = WindowsBuildHelper(self, prefix)
+            build_helper = WindowsBuildHelper(self)
         else:
-            build_helper = UnixLikeBuildHelper(self, prefix)
-        build_helper.build_embedded()
+            build_helper = UnixLikeBuildHelper(self)
+        build_helper.build(prefix)
+        build_helper.install(prefix)
 
         if not self.options.packages:
             return
 
-        build_helper.enable_site_packages()
-        bootstrap = build_helper.build_bootstrap()
+        build_helper.enable_site_packages(prefix)
+        bootstrap = build_helper.build_bootstrap(prefix)
```

Comment:
Why pass the prefix individually? Some of the methods don't even use it depending on the OS (e.g. for `WindowsBuildHelper.build_bootstrap()` it's an unused arg). I understand the split into source/build/install and it's a welcome improvement. The prefix move just feels confusing.